packages/webauthn/index.ts

```diff
@@ -0,0 +1,196 @@
+  )[0] as bigint
+  console.log(
+    `[webauthn][debug]\nuserDisplayName: ${userDisplayName}\nchallengeBase64Url: ${challengeBase64Url}\ncredPubKeyXHex: ${credPubKeyXHex}\ncredPubKeyYHex: ${credPubKeyYHex}`
+  )
+  return {
+    origin: origin,
+    credentialId: credIdBase64Url,
+    publicKeyX: credPubKeyXUint256,
+    publicKeyY: credPubKeyYUint256
+  } as WebauthnRegistration
+}
+
+export const requestWebauthn = async (
+  params: WebauthnRequest
+): Promise<WebauthnAuthentication> => {
+  const authResJson = await startAuthentication({
+    ...(params.credentialId
+      ? {
+          allowCredentials: [
+            { id: params.credentialId, type: defaultWebauthn.pubKeyCredType }
+          ] as PublicKeyCredentialDescriptorJSON[]
+        }
+      : {}),
+    userVerification: defaultWebauthn.userVerificationRequirement,
+    challenge: params.challenge
+  } as PublicKeyCredentialRequestOptionsJSON)
+
+  const authDataUrlB64 = authResJson.response.authenticatorData
+  const authDataHex = `0x${isoUint8Array.toHex(
+    isoBase64URL.toBuffer(authDataUrlB64)
+  )}`
+  const clientDataJsonUrlB64 = authResJson.response.clientDataJSON
+  const clientDataJsonUtf8 = isoBase64URL.toString(clientDataJsonUrlB64)
+  const sigUrlB64 = authResJson.response.signature
+  const [sigRUint, sigSUint] = parseSignature(sigUrlB64)
+  //   console.log(`[Test][P256]`)
+
+  //   console.log(`[Test][P256] r: ${p256Sig.r}`)
+  //   console.log(`[Test][P256] s: ${p256Sig.s}`)
+  return {
+    authenticatorData: authDataHex,
+    clientDataJson: clientDataJsonUtf8,
+    sigantureR: sigRUint,
+    signatureS: sigSUint
+  } as WebauthnAuthentication
+}
+
+export const defaultWebauthn = {
+  attestationConveyancePreference: "none" as AttestationConveyancePreference,
+  // This Relying Party will accept either an ES256 or RS256 credential, but prefers an ES256 credential.
+  pubKeyCredAlgEs256: -7 as COSEAlgorithmIdentifier, // ES256 (Webauthn's default algorithm)
+  pubKeyCredAlgRs256: -257 as COSEAlgorithmIdentifier, // RS256 (for Windows Hello and others)
+  // Try to use UV if possible. This is also the default.
+  pubKeyCredType: "public-key" as PublicKeyCredentialType,
+  residentKeyRequirement: "required" as ResidentKeyRequirement,
+  userVerificationRequirement: "required" as UserVerificationRequirement,
+  timeout: 300000, // 5 minutes
+  // Relying Party
+  rpName: "Waallet Extension",
+  // Make excludeCredentials check backwards compatible with credentials registered with U2F
+  extensions: {
+    largeBlob: {
+      support: "preferred" // "required", "preferred"
+    }
+  },
+  requireResidentKey: true
+}
+
+const parseSignature = (signature: UrlB64String): [bigint, bigint] => {
+  if (!isoBase64URL.isBase64url(signature)) {
+    console.log(`${signature} is not Base64Url`)
+    return [BigInt(0), BigInt(0)] as const
+  }
+  const n = BigInt(
+    "0xFFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551"
+  )
```

Comment:
nit: `n` could be derived from `p256.CURVE.n`

https://github.com/pilagod/waallet-extension/blob/4229422abb41fc2731b7086926bd7b0dc8cf886a/packages/account/PasskeyAccount/passkeyOwnerP256.ts#L42-L44